src/utils.ts

```diff
@@ -261,32 +256,25 @@ export function getPlatform(): Platform {
 }
 
 /**
- * Get the path to the first valid browser for the current session
- * The search order is: launchConfig > vscode setting > platform default
- * For each platform the order is: stable > beta > dev > canary
- * For windows we will try: program files > local app data
+ * Gets the browser path for the specified browser flavor.
  * @param config The settings specified by a launch config, if any
  */
-export async function getBrowserPath(config: Partial<IUserConfig> = {}) {
+export async function getBrowserPath(config: Partial<IUserConfig> = {}): Promise<string> {
     const settings = vscode.workspace.getConfiguration(SETTINGS_STORE_NAME);
-    const browserPath = config.browserPath || settings.get("browserPath") || "";
-
-    if (!browserPath) {
-        const platform = getPlatform();
-        const searchPaths = platform === "Windows" ? WIN_MSEDGE_PATHS :
-            platform === "OSX" ? OSX_MSEDGE_PATHS :
-                [];
-
-        // Find the first one that exists
-        for (const p of searchPaths) {
-            if (await fse.pathExists(p)) {
-                return p;
-            }
+    const flavor: BrowserFlavor | undefined = config.browserFlavor || settings.get("browserFlavor");
+
+    switch (getPlatform()) {
+        case "Windows": {
+           return await verifyFlavorPath(flavor, "Windows");
+           break;

```

Comment:
We don't need break statements if we are returning the line before.

Reply:
😱, how did the linter not catch that one?! nice one Michael.